apps/backend/src/libs/types/repository.type.ts

```diff
@@ -1,8 +1,10 @@
+import { type PaginationResponseDto } from "~/libs/types/types.js";
+
 type Repository<T = unknown> = {
 	create(payload: unknown): Promise<T>;
 	delete(id: number): Promise<boolean>;
 	find(id: number): Promise<T | null>;
-	findAll(payload: unknown): Promise<{ items: T[] }>;
+	findAll(payload: unknown): Promise<{ items: T[] } | PaginationResponseDto<T>>;
```

Comment:
```suggestion
	findAll(payload: unknown): Promise<{ items: T[] }>;
```
do not add it here. only for methods where you need it